### PR TITLE
feat: [DDS-62/DDS-85] Embedded verification relationships and more tests

### PR DESCRIPTION
**Pull request by spivachuk**
- Added support for embedded verification relationships.
- Added more tests for verification relationships.

### CODE REVIEW

File: x/did/types/fee.pb.go [hunk 1]
```diff
@@ -81,30 +81,31 @@ func init() {
...
-	0x16, 0xba, 0x57, 0xa2, 0xed, 0x63, 0x5b, 0x3b, 0x9b, 0x78, 0x7c, 0xba, 0xb4, 0xb1, 0xc3, 0xfc,
-	0xcc, 0x5d, 0x76, 0xb5, 0x22, 0x77, 0x46, 0xa4, 0x14, 0xee, 0x51, 0xe7, 0xed, 0xb5, 0x05, 0x33,
-	0xf3, 0x3d, 0xea, 0x98, 0x50, 0x08, 0x0e, 0xa4, 0x5e, 0xf5, 0x16, 0x96, 0xf6, 0x0d, 0xe9, 0x87,
-	0x30, 0x3f, 0xa3, 0x49, 0x19, 0x88, 0x31, 0xa6, 0x28, 0x75, 0x02, 0xff, 0xc4, 0xd6, 0x7c, 0x49,
-	0xe5, 0xe8, 0xa2, 0x51, 0xc1, 0x99, 0x92, 0xc8, 0x0c, 0x67, 0x99, 0xe1, 0x6b, 0xe6, 0x05, 0xa6,
-	0xe2, 0x75, 0x72, 0x97, 0xa0, 0xdb, 0x7f, 0x49, 0x11, 0x58, 0xa7, 0x08, 0x6c, 0x52, 0x04, 0x3e,
-	0x53, 0x04, 0x9e, 0x76, 0x48, 0xdb, 0xec, 0x90, 0xf6, 0xbe, 0x43, 0xda, 0xfd, 0xf9, 0xbe, 0x6d,
-	0xb9, 0x3d, 0x79, 0xb6, 0x02, 0xe6, 0x52, 0xb2, 0x92, 0xab, 0x94, 0xde, 0xed, 0xbf, 0x32, 0xeb,
-	0x8b, 0xef, 0x00, 0x00, 0x00, 0xff, 0xff, 0x20, 0xf3, 0xe8, 0x95, 0xe4, 0x01, 0x00, 0x00,
+	// 374 bytes of a gzipped FileDescriptorProto
```
Comment (Toktar) [inline x/did/types/fee.pb.go]: Why do we have changes in this file?

File: x/did/types/genesis.pb.go [hunk 2]
```diff
@@ -142,28 +142,28 @@ func init() {
...
-	0xaf, 0xf4, 0x49, 0x12, 0xfa, 0xfb, 0xbc, 0xf1, 0xe5, 0x96, 0xcb, 0x09, 0x9a, 0x42, 0x66, 0xb8,
-	0xcc, 0xf4, 0x40, 0xc1, 0xd5, 0x0a, 0xf3, 0xbb, 0xdc, 0x5b, 0x9e, 0x94, 0x1a, 0xa4, 0x55, 0xac,
-	0xf8, 0x2b, 0xf6, 0xe7, 0x9f, 0xa4, 0x5e, 0x7e, 0xef, 0xc6, 0x3f, 0xe1, 0x7c, 0x88, 0x98, 0x16,
-	0x72, 0x2a, 0x27, 0xa6, 0x59, 0x0d, 0x58, 0xa7, 0x1e, 0xef, 0xae, 0x1b, 0x9c, 0x23, 0x5e, 0x5b,
-	0x3a, 0xa9, 0x0d, 0x57, 0xeb, 0x59, 0xf7, 0x6d, 0x2e, 0xd8, 0x6c, 0x2e, 0xd8, 0xc7, 0x5c, 0xb0,
-	0x97, 0x85, 0xa8, 0xcc, 0x16, 0xa2, 0xf2, 0xbe, 0x10, 0x95, 0xbb, 0xc3, 0x91, 0xa2, 0xf1, 0xe3,
-	0x20, 0xcc, 0xf4, 0x24, 0x72, 0x2d, 0xdb, 0x79, 0x94, 0x6b, 0xc0, 0xe8, 0xc9, 0x56, 0x4e, 0xcf,
-	0x05, 0x9a, 0xc1, 0x7f, 0x5b, 0xf9, 0xf1, 0x67, 0x00, 0x00, 0x00, 0xff, 0xff, 0xed, 0xeb, 0xc7,
-	0x91, 0xd1, 0x01, 0x00, 0x00,
+	// 328 bytes of a gzipped FileDescriptorProto
```
Comment (Toktar) [inline x/did/types/genesis.pb.go:145]: Why do we have changes in this file?

File: x/did/types/query.pb.go [hunk 3]
```diff
@@ -318,35 +318,36 @@ func init() {
...
-	0xef, 0xbe, 0xfc, 0xf8, 0x60, 0xdc, 0x45, 0x65, 0x7c, 0xe0, 0xdb, 0xc2, 0x4b, 0x46, 0x56, 0xe8,
-	0x23, 0x80, 0xb7, 0x72, 0xc5, 0x50, 0xf3, 0x98, 0x75, 0x7e, 0xbe, 0x56, 0xeb, 0x24, 0x56, 0xa7,
-	0xe9, 0xa8, 0x34, 0x6d, 0xd4, 0x2a, 0x4c, 0x83, 0xf5, 0xc1, 0xe0, 0xa5, 0xbe, 0x59, 0xa1, 0x4f,
-	0x00, 0x96, 0x0b, 0xcf, 0x1e, 0x75, 0x0f, 0xef, 0xff, 0xb7, 0x41, 0x5b, 0x8f, 0xfe, 0x59, 0xa7,
-	0x3b, 0x34, 0x55, 0x87, 0x1a, 0x72, 0x8a, 0x3b, 0x4c, 0xb5, 0xe6, 0x49, 0xef, 0xf3, 0xc6, 0x06,
-	0xeb, 0x8d, 0x0d, 0xbe, 0x6f, 0x6c, 0xf0, 0x7e, 0x6b, 0x97, 0xd6, 0x5b, 0xbb, 0xf4, 0x75, 0x6b,
-	0x97, 0x5e, 0xd7, 0xc7, 0x4c, 0x4e, 0x92, 0x91, 0x1b, 0xf0, 0xa9, 0xf6, 0x51, 0xd7, 0x76, 0xc4,
-	0x09, 0xc5, 0x6f, 0x95, 0xa9, 0x5c, 0xc4, 0x54, 0x8c, 0x2e, 0xd4, 0x1f, 0xde, 0xf9, 0x19, 0x00,
-	0x00, 0xff, 0xff, 0x99, 0xb4, 0x16, 0xb9, 0x44, 0x04, 0x00, 0x00,
+	// 449 bytes of a gzipped FileDescriptorProto
```
Comment (Toktar) [inline x/did/types/query.pb.go:321]: Why do we have changes in this file?

File: x/did/types/v1/common.pb.go [hunk 4]
```diff
@@ -81,19 +81,19 @@ func init() {
 func init() { proto.RegisterFile("cheqd/did/v1/common.proto", fileDescriptor_488ad3ab822f2393) }
 
 var fileDescriptor_488ad3ab822f2393 = []byte{
-	// 178 bytes of a gzipped FileDescriptorProto
```
Comment (Toktar) [inline x/did/types/v1/common.pb.go:84]: Why do we have changes in v1 files?

**Review by Toktar — COMMENTED** [review]
Thanks for a great job!
Just a few questions:
- Can we remove changes in proto generated files where it's not needed?
- Does this PR contain changes for DID Resource validation?
- Do we have a unit test where VerificationRelationship contains both ID and embedded VM (invalid case)?
- Could you please make a merge with `develop` brunch?

**Comment by spivachuk** [comment]
@Toktar Below please find answers to the questions from your previous comment:

>Can we remove changes in proto generated files where it's not needed?

Agreed to keep changes in these files "as is".

>Does this PR contain changes for DID Resource validation?

It contains changes in `DidDoc` validation. No changes are needed in `Resource` validation.

>Do we have a unit test where VerificationRelationship contains both ID and embedded VM (invalid case)?

Yes, it is [here](https://github.com/canow-co/cheqd-node/blob/877e5be754b022dd4218260e9efe2661a3604f67/x/did/types/diddoc_verification_relationship_test.go#L74).

>Could you please make a merge with `develop` brunch?

Done.

**Review by Toktar — APPROVED** [review]

**Pull request merged**